tests_and_analysis/test/script_tests/utils.py

```diff
@@ -0,0 +1,125 @@
+        os.mkdir(dos_or_dispersion_data_folder)
+    filepath_prefix = os.path.join(dos_or_dispersion_data_folder, filename_prefix)
+    return filepath_prefix
+
+
+def get_dispersion_data_filepath_prefix() -> str:
+    """
+    Returns
+    -------
+    str: a full path prefix of all dispersion regression test files.
+    """
+    return _get_dos_or_dispersion_data_filepath_prefix(get_dispersion_data_folder())
+
+
+def get_dos_data_filepath_prefix() -> str:
+    """
+    Returns
+    -------
+    str: a full path prefix of all dos regression test files.
+    """
+    return _get_dos_or_dispersion_data_filepath_prefix(get_dos_data_folder())
+
+
+def _iter_regression_test_files(filepath_prefix: str, data_folder: str):
+    # Loop through files in data_folder
+    for file in os.listdir(data_folder):
+        file_full_path = os.path.join(data_folder, file)
+        # Only accept those that start with filepath_prefix
+        if file_full_path.startswith(filepath_prefix):
+            filenum = file_full_path[len(filepath_prefix):][0]
+            # Return the number used in the file and the full path to the file
+            yield int(filenum), file_full_path
+
+
+def iter_dispersion_data_files():
+    """
+    A generator function to get all the files
+     containing regression testing data for scripts/dispersion.py
+    Yields the number of the file and the full file path as a tuple.
+
+    Examples
+    --------
+         >>> for filenum, file in iter_dispersion_data_files():
+         >>>   print(filenum)
+         >>>
+         >>> 0
+         >>> 1
+    """
+    filepath_prefix = get_dispersion_data_filepath_prefix()
+    data_folder = get_dispersion_data_folder()
+    yield from _iter_regression_test_files(filepath_prefix, data_folder)
+
+
+def iter_dos_data_files():
+    """
+    A generator function to get all the files
+     containing regression testing data for scripts/dos.py
+    Yields the number of the file and the full file path as a tuple.
+
+    Examples
+    --------
+         >>> for filenum, file in iter_dos_data_files():
+         >>>   print(filenum)
+         >>>
+         >>> 0
+         >>> 1
+    """
+    filepath_prefix = get_dos_data_filepath_prefix()
+    data_folder = get_dos_data_folder()
+    yield from _iter_regression_test_files(filepath_prefix, data_folder)
+
+
+def write_plot_to_file_for_regression_testing(filepath_prefix: str):
+    """
+    Get the current matplotlib plot with gcf() and write the xydata of the lines of the plot
+     to separate csv files using the filepath_prefix parameter as a prefix and the
+      index of the line and .csv to complete the filename.
+
+    Parameters
+    ----------
+    filepath_prefix : str
+        A prefix to use to write the files e.g. C:/data
+    """
+    for index, line in enumerate(matplotlib.pyplot.gcf().axes[0].lines):
+        np.savetxt(filepath_prefix + str(index) + ".csv", line.get_xydata().T, delimiter=",")
```

Comment:
Is there a reason why each line is written to a separate .csv? I feel like it would be easier to keep track of the data if it was all in the same file, but maybe there's a good reason for doing it this way?

Reply:
Though this would make sense, we cannot save a 3D array in human readable form using numpy, so it is best to stick with this for now